### PR TITLE
Improve documentation and various flake8 linting

### PR DESCRIPTION
**Pull request by steven-murray**
This PR adds a bunch of working flake8 tests, especially ones that check documentation.

It also fixes links in docstrings to other classes/functions.

It also puts all cosmoHammer stuff into one file, which makes it easier to look at in documentation (and it's not massive anyway).

### CODE REVIEW

File: AUTHORS.rst [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
+Here follows a list of individuals who have contributed in any way to ``21CMMC``:
 
-
-Contributors
-============
+* `Brad Greig <https://github.com/BradGreig>`_
+* `Steven Murray <https://github.com/steven-murray>`_
+* `Andrei Mesinger <https://github.com/andreimesinger>`_
+* `Catherine Watkinson <https://github.com/catherinewatkinson>`_
+* `Yuxiang Qin <https://github.com/qyx268>`_
```
Comment (BradGreig) [inline AUTHORS.rst:15]: You should probably add Jaehong (he has contributed things to 21CMMC). Probably should add Tom as well

File: devel/valgrind_filtered.out [hunk 2]
```diff
@@ -0,0 +1,120 @@
+==8849== 232 bytes in 1 blocks are possibly lost in loss record 27,598 of 35,346
```
Comment (BradGreig) [inline devel/valgrind_filtered.out]: This seems to be an output file. Is this here for any particular reason? Does it actually serve any purpose?

File: src/py21cmmc/cosmoHammer.py [hunk 3]
```diff
@@ -0,0 +1,1011 @@
...
+                return False
+            for j, val in enumerate(v):
+                if val != other.values[i][j]:
+                    return False
+
+        return True
+
+
+class LikelihoodComputationChain(_Chain):
+    """Feature-laded replacement of :class:`cosmoHammer.LikelihoodComputationChain`."""
```
Comment (BradGreig) [inline src/py21cmmc/cosmoHammer.py]: Typo? laden?

File: src/py21cmmc/cosmoHammer.py [hunk 4]
```diff
@@ -0,0 +1,1011 @@
...
+        -------
+        ctx : dict-like
+            A filled context object containing all model quantities generated by core modules in
+            the chain.
+
+        Notes
+        -----
+        The data generated by this method should ideally be *deterministic*, so that input
+        parameters map uniquely to
+        output data. All data necessary to full evaluate probabilities of mock data from the
```
Comment (BradGreig) [inline src/py21cmmc/cosmoHammer.py]: typo: fully

**Review by BradGreig — APPROVED** [review]
Essentially I only read most of the comments etc. for the documentation. I assume other stuff has been tested and functions correctly

File: devel/valgrind_filtered.out (hunk 2 above)
Comment (steven-murray) [reply devel/valgrind_filtered.out]: Nope, must've been added accidentally. I'll remove it.

File: AUTHORS.rst (hunk 1 above)
Comment (steven-murray) [reply AUTHORS.rst:15]: Thanks, will add them in.

File: src/py21cmmc/cosmoHammer.py (hunk 3 above)
Comment (steven-murray) [reply src/py21cmmc/cosmoHammer.py]: Woops.

File: src/py21cmmc/cosmoHammer.py (hunk 4 above)
Comment (steven-murray) [reply src/py21cmmc/cosmoHammer.py]: Got it, thanks.

**Pull request merged**